feat: 'Typed Transaction Envelope' added to transaction page (#3836)
ethereum/ethereum-org-website: src/content/developers/docs/transactions/index.md

<iframe width="100%" height="315" src="https://www.youtube.com/embed/er-0ihqFQB0" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

## Typed Transaction Envelope

> Ethereum has different transaction types that specify what a transaction should do. For example, sending Ether to an address, deploying a contract, etc.  

Reviewer: This is a little inaccurate -- all tx types (so far) can send ether, deploy contracts, etc. These functions are considered different "types" of transactions. The different "types" have much more to do with allowing new transaction behavior that doesn't exist in the protocol already (e.g. access lists).

Author: Hey @lightclient, can you help me reframe or replace this statement?
Thanks!!!

Reviewer: Maybe something like "Over time Ethereum has evolved to support multiple types of transactions. This allows new features like access lists and EIP-1559 to be implemented without affecting legacy transaction formats."?

Author: yeah that looks great!!
Thanks !

> Ethereum now has a new transaction standard that was defined and created by Micah Zoltu in [EIP-2718](https://eips.ethereum.org/EIPS/eip-2718): Typed Transaction Envelope, which forms the basis for some new and other yet-to-be-developed features on Ethereum.

Reviewer 2: Recommend against using temporally sensitive terminology in documentation like "now has" and "that was".  The website should speak in terms of what is, largely ignoring what was unless explicitly discussing history.

> Ethereum now has a new transaction standard that was defined and created by Micah Zoltu in [EIP-2718](https://eips.ethereum.org/EIPS/eip-2718): Typed Transaction Envelope, which forms the basis for some new and other yet-to-be-developed features on Ethereum.

Reviewer 2: Referring to the author of the EIP doesn't add value, so I recommend removing it here.  The EIP has an author line in the header if people are interested in that kind of history.

Reviewer 2: Recommend the following style, where the title is part of the link.
```
[EIP-2718: Typed Transaction Envelope](https://eips.ethereum.org/EIPS/eip-2718)
```

Reviewer: If *must* refer to the author of the EIP, I recommend using their full title.

```
Micah Zoltu, Overlord of Ethereum Governance
```

> Ethereum now has a new transaction standard that was defined and created by Micah Zoltu in [EIP-2718](https://eips.ethereum.org/EIPS/eip-2718): Typed Transaction Envelope, which forms the basis for some new and other yet-to-be-developed features on Ethereum.

Reviewer 2: Recommend the following style, where the title is part of the link.
```
[EIP-2718: Typed Transaction Envelope](https://eips.ethereum.org/EIPS/eip-2718)
```

> Previously, Ethereum had one format for transactions. Each transaction consists of a nonce, gas price, gas limit, to address, value, data, v, r, and s. These fields are RLP-encoded, to look something like this:  

Reviewer 2: As above, recommend removing the temporally sensitive wording and focusing on the current state of things, which is that we have 4 transaction types now:
Cross Chain Legacy
Chain Locked Legacy [EIP-155](https://eips.ethereum.org/EIPS/eip-155)
Chain Locked with Access Lists [EIP-2930](https://eips.ethereum.org/EIPS/eip-2930)
Chain Locked with Split Fees [EIP-1559](https://eips.ethereum.org/EIPS/eip-1559)

`RLP([nonce, gasPrice, gasLimit, to, value, data, v, r, s])`

> EIP-2718 defines a new generalised envelope for typed transactions. In the new standard, transactions look like this:  

Reviewer: ```suggestion
EIP-2718 defines a new generalised envelope for typed transactions. In the new standard, transactions are interpreted as:  
```

> ## Typed Transaction Envelope {#typed-transaction-envelope}
> 
> Over time, Ethereum has evolved to support multiple types of transactions. This allows new features like access lists and EIP-1559 to be implemented without affecting legacy transaction formats.  
> [EIP-2718: Typed Transaction Envelope](https://eips.ethereum.org/EIPS/eip-2718) defines a transaction type that is an envelope for future transaction types.
> 
> Previously, Ethereum had one format for transactions. Each transaction consists of a nonce, gas price, gas limit, to address, value, data, v, r, and s. These fields are RLP-encoded, to look something like this:  
> `RLP([nonce, gasPrice, gasLimit, to, value, data, v, r, s])`
> 
> EIP-2718 is a new generalised envelope for typed transactions. In the new standard, transactions are interpreted as:  
> `TransactionType || TransactionPayload`
> 
> Where the fields are defined as:
> 
> - TransactionType: a number between 0 and 0x7f, for a total of 128 possible transaction types.
> - TransactionPayload: an arbitrary byte array, defined by the transaction type.
> 

Reviewer 3: ```suggestion
## Typed Transaction Envelope {#typed-transaction-envelope}

Ethereum originally had one format for transactions. Each transaction consists of a nonce, gas price, gas limit, to address, value, data, v, r, and s. These fields are RLP-encoded, to look something like this:  

`RLP([nonce, gasPrice, gasLimit, to, value, data, v, r, s])`

Ethereum has evolved to support multiple types of transactions to allow for new features such as access lists and [EIP-1559](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1559.md) to be implemented without affecting legacy transaction formats.  

[EIP-2718: Typed Transaction Envelope](https://eips.ethereum.org/EIPS/eip-2718) defines a transaction type that is an envelope for future transaction types.

EIP-2718 is a new generalised envelope for typed transactions. In the new standard, transactions are interpreted as:  

`TransactionType || TransactionPayload`

Where the fields are defined as:

- `TransactionType` - a number between 0 and 0x7f, for a total of 128 possible transaction types.
- `TransactionPayload` an arbitrary byte array defined by the transaction type.

```

> ## Further reading {#further-reading}
> 
> _Know of a community resource that helped you? Edit this page and add it!_

Reviewer 3: ```suggestion
## Further reading {#further-reading}

- [EIP-2718: Typed Transaction Envelope](https://eips.ethereum.org/EIPS/eip-2718) 

_Know of a community resource that helped you? Edit this page and add it!_
```

<iframe width="100%" height="315" src="https://www.youtube.com/embed/er-0ihqFQB0" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

## Typed Transaction Envelope {#typed-transaction-envelope}

> Ethereum originally had one format for transactions. Each transaction consists of a nonce, gas price, gas limit, to address, value, data, v, r, and s. These fields are RLP-encoded, to look something like this:  

Reviewer 3: ```suggestion
Ethereum originally had one format for transactions. Each transaction contained a nonce, gas price, gas limit, to address, value, data, v, r, and s. These fields are RLP-encoded, to look something like this:  
```

`RLP([nonce, gasPrice, gasLimit, to, value, data, v, r, s])`

Ethereum has evolved to support multiple types of transactions to allow for new features such as access lists and [EIP-1559](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1559.md) to be implemented without affecting legacy transaction formats.  

[EIP-2718: Typed Transaction Envelope](https://eips.ethereum.org/EIPS/eip-2718) defines a transaction type that is an envelope for future transaction types.

EIP-2718 is a new generalised envelope for typed transactions. In the new standard, transactions are interpreted as:  

`TransactionType || TransactionPayload`

Where the fields are defined as:

- `TransactionType` - a number between 0 and 0x7f, for a total of 128 possible transaction types.

> - `TransactionPayload` an arbitrary byte array defined by the transaction type.

Reviewer 3: ```suggestion
- `TransactionPayload` - an arbitrary byte array defined by the transaction type.
```
